lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescaleset.py

```diff
@@ -453,7 +513,9 @@ def __init__(self):
             overprovision=dict(type='bool', default=True),
             single_placement_group=dict(type='bool', default=True),
             zones=dict(type='list'),
-            custom_data=dict(type='str')
+            custom_data=dict(type='str'),
+            plan=dict(type='dict'),
```

Comment:
You should add the suboptions here as well. See https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/docker/docker_container.py#L3311-L3317 for an example.

Reply:
If I change that to 
_plan=dict(type='dict', options=dict(publisher=dict(type='str'),product=dict(type='str'),name=dict(type='str'))),_
how do I specify that the name, product and publisher are required if a plan is used at all?

Reply:
Add `required=True` to `name`, `pulisher` and `product`. As long as you don't add it to `plan`, they will only be required if `plan` is actually specified.

Reply:
ok, this will be in my next push